File: modules/core/src/encoder/SegmentType.ts
/**
 * 0 = default - reserved for core string (does not need to be present in the core string)
 * 1 = OOB vendors disclosed
 * 2 = OOB vendors allowed
 * 3 = PublisherTC
 */
import {Segments} from '../model';

export class SegmentType {

  public static readonly '0': string = Segments.core;
  public static readonly '1': string = Segments.vendorsDisclosed;
  public static readonly '2': string = Segments.vendorsAllowed;
  public static readonly '3': string = Segments.publisherTC;

  public static readonly core: number = 0;
  public static readonly vendorsDisclosed: number = 1;
  public static readonly vendorsAllowed: number = 2;
  public static readonly publisherTC: number = 3;

  public static readonly numTypes: number = 4;

};

File: GDPR-Phase2-Java/src/main/java/com/gdpr/phase2/encoder/segment/SegmentType.java
package com.gdpr.phase2.encoder.segment;

import com.gdpr.phase2.model.Segments;
import lombok.*;

@Getter @Setter
public class SegmentType {
    private SegmentType() {
    }
    private static  SegmentType instance = new SegmentType();
    public static SegmentType getInstance() {
        return instance;
    }

    public static final String zero = Segments.core;
    public static final String one = Segments.vendorsDisclosed;
    public static final String two = Segments.vendorsAllowed;
    public static final String three = Segments.publisherTC;

    public static final int core = 0;
    public static final int vendorsDisclosed = 1;
    public static final int vendorsAllowed = 2;
    public static final int publisherTC = 3;

    public static final int numTypes = 4;
}
